### PR TITLE
Implement PodsList Page with User-Selectable Broker Pod for Jolokia Endpoint Access

### PR DESCRIPTION
**Pull request by Msarawan**
Designed brokers pods page to list set of pods created within specific broker. 

Pods List Page:
Screenshot from 2024-05-02 21-49-33

Broker Details Page:
Screenshot from 2024-05-06 14-36-10

### CODE REVIEW

**Review by lavocatt — COMMENTED** [review]
Good work! I think this is a great implementation of the requested design :rocket:. The only thing left to do IMO is to improve on the commit structure.

I think the contribution in itself is focused enough to be contained in a single commit.

For the title of the commit I would go for something like: `brokerDetails: add an indirection level to choose a pod` or better if you get inspired!

As for the description, I think it would be great to be a bit verbose on describing the new workflow for the user to go from brokers list to broker details. Also mentioning Jolokia feels important.

Something along those lines:

> Before, the frontend was limited to only connect to the jolokia endpoint of the first pod. It would have meant limitations to what information could be displayed in the future.
> As a result, this commit changes the steps a user needs to take in order to display the broker details. We now request that the user selects a pod prior to accessing the page.
> This change paves the way for displaying pod-related information such as number of messages in flight in the queues.

> [!TIP]
> How to squash a branch into a single commit
> ```
> git checkout broker-pods-page-design
> git pull # don't forget to pull to get my commit in
> git rebase main broker-pods-page-design -i
> ```
> The last command will open a text editor in which you can specify what to do with every commits. Replace `pick` by `squash` for all the commits except the first one. Git will squash the all together asking you to review the commit message at each step. On the last squashing step (number 7), update the commit message with a title and a description.

**Comment by Msarawan** [comment]
> Good work! I think this is a great implementation of the requested design 🚀. The only thing left to do IMO is to improve on the commit structure.
> 
> I think the contribution in itself is focused enough to be contained in a single commit.
> 
> For the title of the commit I would go for something like: `brokerDetails: add an indirection level to choose a pod` or better if you get inspired!
> 
> As for the description, I think it would be great to be a bit verbose on describing the new workflow for the user to go from brokers list to broker details. Also mentioning Jolokia feels important.
> 
> Something along those lines:
> 
> > Before, the frontend was limited to only connect to the jolokia endpoint of the first pod. It would have meant limitations to what information could be displayed in the future.
> > As a result, this commit changes the steps a user needs to take in order to display the broker details. We now request that the user selects a pod prior to accessing the page.
> > This change paves the way for displaying pod-related information such as number of messages in flight in the queues.
> 
> Tip
> 
> How to squash a branch into a single commit
> 
> ```
> git checkout broker-pods-page-design
> git pull # don't forget to pull to get my commit in
> git rebase main broker-pods-page-design -i
> ```
> 
> The last command will open a text editor in which you can specify what to do with every commits. Replace `pick` by `squash` for all the commits except the first one. Git will squash all the commits together and will ask you to update the commit message at each step. On the last squashing step (number 7 if I'm guessing correctly), update the commit message with the aforementioned title and description.

Thanks @lavocatt !

**Review by lavocatt — APPROVED** [review]
All good to me! Good work!

**Comment by Msarawan** [comment]
> All good to me! Good work!

Thanks! @lavocatt

File: src/brokers/broker-details/BrokerDetails.container.tsx [hunk 1]
```diff
@@ -92,9 +106,9 @@ const BrokerDetailsPage: FC = () => {
         className="pf-c-page__main-tabs"
       >
         <div className="pf-u-mt-md pf-u-ml-md pf-u-mb-md">
-          <BrokerDetailsBreadcrumb name={name} namespace={namespace} />
+          <BrokerDetailsBreadcrumb name={brokerName} namespace={namespace} />
           <Title headingLevel="h2">
-            {t('broker')} {name}
+            {t('pod')} {podName}
```
[the remark this answers is not in the document]
Comment (lavocatt) [reply src/brokers/broker-details/BrokerDetails.container.tsx:111]: But we can also improve on that later on.

[the remark this answers is not in the document]
Comment (Msarawan) [reply src/brokers/broker-details/BrokerDetails.container.tsx:111]: > I think we could have a title that both reflects the broker and the pod:
> 
> {t('broker')} {brokerName} {t('/')} {podName}}
> 
> Wdyt?

yeah, make sense to me. anyways my motive is to upon onclick on each pod that should display the selected pod-name on broker-details page. but with your suggestions we can do changes in the title with both broker name followed by it's selected pod name.

let me i will update the modification.

**Review by lavocatt — APPROVED** [review]
:rocket:

**Pull request merged**